### PR TITLE
Capz templates

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/18620

This PR adds initial support to generate templates for CAPZ clusters.
I don't consider this a final result, but a starting point to gather early feedback about the approach.

### CODE REVIEW

File: cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    "release.giantswarm.io/version": "{{ .Version }}"
+    "giantswarm.io/cluster": "{{ .Name }}"
+    "cluster.x-k8s.io/cluster-name": "{{ .Name }}"
+    "giantswarm.io/organization": "{{ .Owner }}"
+    "cluster.x-k8s.io/watch-filter": "capi"
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+spec:
+  location: ""
```
Comment (MarcelMue) [inline cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl:13]:
```suggestion
```

File: cmd/template/cluster/provider/templates/azure/azure_machine_template.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  template:
+    spec:
+      dataDisks:
+      - diskSizeGB: 256
+        lun: 0
+        nameSuffix: etcddisk
+      osDisk:
+        diskSizeGB: 128
+        osType: Linux
+      sshPublicKey: ""
```
Comment (MarcelMue) [inline cmd/template/cluster/provider/templates/azure/azure_machine_template.yaml.tmpl:22]:
```suggestion
```

File: cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl (hunk 1 above)
Comment (whites11) [reply cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl:13]: Can't remove it. `apply` fails with validation error.

File: cmd/template/cluster/provider/templates/azure/azure_machine_template.yaml.tmpl (hunk 2 above)
Comment (whites11) [reply cmd/template/cluster/provider/templates/azure/azure_machine_template.yaml.tmpl:22]: Can't remove it. `apply` fails with validation error.

File: cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl (hunk 1 above)
Comment (whites11) [reply cmd/template/cluster/provider/templates/azure/azure_cluster.yaml.tmpl:13]:
```
error: error validating "STDIN": error validating data: ValidationError(AzureCluster.spec): missing required field "location" in io.x-k8s.cluster.infrastructure.v1alpha4.AzureCluster.spec; if you choose to ignore these errors, turn validation off with --validate=false
```
Despite the defaulting

**Review by MarcelMue — APPROVED** [review]
god dang crd validation

**Pull request merged**